homeassistant/components/traccar_server/coordinator.py

```diff
@@ -0,0 +1,161 @@
+
+        if TYPE_CHECKING:
+            assert isinstance(devices, list[DeviceModel])  # type: ignore[misc]
+            assert isinstance(positions, list[PositionModel])  # type: ignore[misc]
+            assert isinstance(geofences, list[GeofenceModel])  # type: ignore[misc]
+
+        for position in positions:
+            if (device := get_device(position["deviceId"], devices)) is None:
+                continue
+
+            attr = {}
+            skip_accuracy_filter = False
+
+            for custom_attr in self.custom_attributes:
+                attr[custom_attr] = getattr(
+                    device["attributes"],
+                    custom_attr,
+                    getattr(position["attributes"], custom_attr, None),
+                )
+                if custom_attr in self.skip_accuracy_filter_for:
+                    skip_accuracy_filter = True
+
+            accuracy = position["accuracy"] or 0.0
+            if (
+                not skip_accuracy_filter
+                and self.max_accuracy > 0
+                and accuracy > self.max_accuracy
+            ):
+                LOGGER.debug(
+                    "Excluded position by accuracy filter: %f (%s)",
+                    accuracy,
+                    device["id"],
+                )
+                continue
+
+            data[device["uniqueId"]] = {
+                "device": device,
+                "geofence": get_first_geofence(
+                    geofences,
+                    position["geofenceIds"] or [],
+                ),
+                "position": position,
+                "attributes": attr,
+            }
+
+        if self.events:
+            self.hass.async_create_task(self.import_events(devices))
+
+        return data
+
+    async def import_events(self, devices: list[DeviceModel]):
+        """Import events from Traccar."""
+        start_time = dt_util.utcnow().replace(tzinfo=None)
+        end_time = None
+
+        if self._last_event_import is not None:
+            end_time = start_time - (start_time - self._last_event_import)
+
+        events = await self.client.get_reports_events(
+            devices=[device["id"] for device in devices],
+            start_time=start_time,
+            end_time=end_time,
+            event_types=self.events,
+        )
+        if not events:
+            return
+
+        self._last_event_import = start_time
+        for event in events:
+            device = get_device(event["deviceId"], devices)
+            self.hass.bus.async_fire(
+                f"traccar_{EVENTS[event['type']]}",
+                {
+                    "device_traccar_id": event["deviceId"],
+                    "device_name": getattr(device, "name", None),
+                    "type": event["type"],
+                    "serverTime": event["eventTime"],
+                    "attributes": event["attributes"],
+                },
+            )
```

Comment:
Wouldn't it make more sense to not implement this and move to event entities after this?

Reply:
That would be a breaking change. So no.

Reply:
This isn't implemented yet right?

Reply:
It is.

Reply:
Can you maybe elaborate a bit more on how you expect everything to happen? They have to setup this intergation right?

Reply:
https://github.com/home-assistant/core/blob/dev/homeassistant/components/traccar/device_tracker.py This YAML configuration is going to be imported to this integration.